### PR TITLE
Development

### PR DESCRIPTION
**Pull request by habibullah791**
ADD .gitignore, testcase and CHANGE the file structure

### CODE REVIEW

**Review by mohdanss — CHANGES_REQUESTED** [review]
Update .gitignore to ignore all the files, like __pycache__ etc.

File: DC-322-L-1 Web Engineering Lab/main.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import utils
+
+
+def registerAccount():
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/main.py:4]: Make sure to follow consistency, python supoorts single and double quotes for string. But adhere to use any of it as a sign of clean and consistent code. 

- Either use '' or "", all across your code.
- Use @ as a sign of readability in docstring
- mention parameters & return type in your functions. (as in example below!)
- You mostly mentioned return in docstring as none or None, be consistent, and follow what's suppoorted in python, its 'None'.

A sample docstring can be like this:
```
def sum(a: int, b: int) -> int:
    ''' 
        @param a: first number
        @param b: second number
        @desc: sum of two numbers
        @return: sum of a and b
    '''
    return a + b
```

```
def printSum(a: int, b: int) -> None:
    ''' 
        @param a: first number
        @param b: second number    

        @description: print sum of two numbers
        @return: None
    '''
    print(sum(a, b))
```

File: DC-322-L-1 Web Engineering Lab/main.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        It checks if the entered account number already exists. If it does, a message is displayed.
+        If the account number is unique, the function creates a new account with an initial balance of 100
+        and stores the account details in a data file.
+
+        Returns:
+        None
+        """
+
+    accountDetails = []
+    utils.registerHeader()
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/main.py:18]: There should be a level of abstraction.
Use 
from utils import * 
from utils import -some specific function/class etc.

Just call it by name. No need to mention utils. everywhere.

File: DC-322-L-1 Web Engineering Lab/main.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+def depositAmmount(accountNumber):
+    """
+        Allows a user to Deposit his/her funds.
+
+        This function will take account number as an argument
+        read data from file and on through iterating that data
+        and check if the withdraw ammount is equal or greater than
+        the balance then upate the object.
+
+        Returns:
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/main.py:89]: wouldn't it be better if:
returns:
    atmData: array user dictionaries, that has updated data after the amouunt has been deposited.

File: DC-322-L-1 Web Engineering Lab/main.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+    atmData = utils.readDataFromFile()
+
+    depositAmmount = utils.validateUserInput(
+        0, 99999999, "Enter Amount to withdraw: ")
+
+    for account in atmData:
+        if accountNumber == account["account_number"]:
+            account_balance = int(account["account_balance"])
+            account["account_balance"] = str(
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/main.py:101]: avoid such indentations, which create confusion, keep it at the same line. CHANGE REST OF SUCH INCONSISTENCIES TOO.

File: DC-322-L-1 Web Engineering Lab/utils.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+        This function will read data from file and store the 
+        data in a list of dictionary and then return it 
+
+        Return:
+        atmData
+    """
+    atmData = []
+
+    with open('atmData.txt', 'r') as file:
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/utils.py:130]: wrap this in try/catch block, what if file doesn't exists?

File: DC-322-L-1 Web Engineering Lab/utils.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        This function will get the list of dictionary 
+        and then convert each dictionary in list and then 
+        in string and overite the existing file it will be 
+        account detail of one new user
+
+        Return:
+        none
+    """
+
+    accountData = [str(item) for item in accountDetails]
```
Comment (mohdanss) [inline DC-322-L-1 Web Engineering Lab/utils.py:159]: Whenever using such statements, which are not clear to beginner, add some comments. - Always write a code, which could be understood by a child wiith the help of comments.

**Review by mohdanss — CHANGES_REQUESTED** [review]

**Comment by mohdanss** [comment]
On each CRUD - file is being updated, I deposit something, it updates the file, I withdraw, it does same. Try something, which can save us from this trouble of costly approach.

HINT: You'll need to read a file at start of the program, and write while existing the system.

**Comment by habibullah791** [comment]
Sure I've updated the project

**Pull request closed without merging**